iOS/DuckDuckGo/Fire/FireConfirmationPresenter.swift

```diff
@@ -0,0 +1,133 @@
+        guard let sheet = hostingController.sheetPresentationController else { return }
+        
+        if #available(iOS 16.0, *) {
+            let targetHeight = calculateContentHeight(for: hostingController.rootView, width: presentingWidth)
+            sheet.detents = [.custom { _ in targetHeight }]
+        } else {
+            sheet.detents = [.large()]
+        }
+        sheet.prefersGrabberVisible = false
+        sheet.preferredCornerRadius = Constants.sheetCornerRadius
+    }
+    
+    private func configureIPadPopoverPresentation(for hostingController: UIHostingController<FireConfirmationView>,
+                                                  from source: AnyObject) {
+        hostingController.modalPresentationStyle = .popover
+        if let source = source as? UIView {
+            hostingController.popoverPresentationController?.sourceView = source
+            hostingController.popoverPresentationController?.sourceRect = source.bounds
+        } else if let source = source as? UIBarButtonItem {
+            hostingController.popoverPresentationController?.barButtonItem = source
+        }
+        let sheetHeight: CGFloat
+        if #available(iOS 16.0, *) {
+            sheetHeight = calculateContentHeight(for: hostingController.rootView, width: Constants.iPadSheetWidth)
+        } else {
+            sheetHeight = Constants.iPadSheetDefaultHeight
+        }
+        hostingController.preferredContentSize = CGSize(width: Constants.iPadSheetWidth, height: sheetHeight)
+    }
+    
+    @available(iOS 16.0, *)
+    private func calculateContentHeight(for view: FireConfirmationView, width: CGFloat) -> CGFloat {
+        let sizingController = UIHostingController(rootView: view)
+        sizingController.disableSafeArea()
+        let targetSize = sizingController.sizeThatFits(in: CGSize(width: width, height: .infinity))
+        return targetSize.height
+    }
+    
+    private func presentLegacyConfirmationAlert(on viewController: UIViewController,
+                                                from source: AnyObject,
+                                                onConfirm: @escaping () -> Void,
+                                                onCancel: @escaping () -> Void) {
+        
+        let alert = ForgetDataAlert.buildAlert(cancelHandler: {
+            onCancel()
+        }, forgetTabsAndDataHandler: {
+            onConfirm()
+        })
+        if let view = source as? UIView {
+            viewController.present(controller: alert, fromView: view)
+        } else if let button = source as? UIBarButtonItem {
+            viewController.present(controller: alert, fromButtonItem: button)
+        } else {
+            assertionFailure("Unexpected sender")
+        }
+    }
+}
+
+private extension FireConfirmationPresenter {
+    enum Constants {
+        static let iPadSheetWidth: CGFloat = 375
+        static let iPadSheetDefaultHeight: CGFloat = 520
+        static let sheetCornerRadius: CGFloat = 12
```

Comment:
We've applied 24 points on Experimental Browsing Menu, It'd be good to double check if 12 is what we want here, for consistency.

Reply:
Good point, started a thread on figma to confirm 👍